### PR TITLE
fix "exists" pre-filter

### PR DESCRIPTION
**Pull request by czue**
https://dimagi-dev.atlassian.net/browse/SAAS-11225

Review by commit.

##### SUMMARY

https://github.com/dimagi/commcare-hq/pull/28313 accidentally neglected to check for operators, so the "exists" filter (which used `!= ''`) was being interpreted as "is empty" which is the exact wrong behavior :man_facepalming: 

##### PRODUCT DESCRIPTION

Fixes the bug which was live for a few days.

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

File: corehq/apps/userreports/reports/filters/values.py [hunk 1]
```diff
@@ -223,6 +223,9 @@ def _has_empty_value(self):
         return self.value['operand'] == '' or self.value['operand'] is None
 
     def _is_empty(self):
+        """
+        Returns true if the value should be treated a filter to show only empty data
```
Comment (millerdev) [inline corehq/apps/userreports/reports/filters/values.py:227]: nit: typo
```suggestion
        Returns true if the value should be treated as a filter to show only empty data
```

File: corehq/apps/userreports/tests/test_report_filters.py [hunk 2]
```diff
@@ -430,7 +430,7 @@ def test_pre_filter_value_exists(self):
             self.assertEqual(filter_value.to_sql_values(), {'exists_field_slug': ''})
             self.assertEqual(
                 str(filter_value.to_sql_filter().build_expression()),
-                'exists_field != :exists_field_slug OR exists_field IS NOT NULL'
+                'exists_field != :exists_field_slug'
```
Comment (millerdev) [inline corehq/apps/userreports/tests/test_report_filters.py:433]: It took a bit of mental exercise for me to reason about why `NULL != ''` happens to do the right thing here. Maybe explicit is better than implicit in this case?

File: corehq/apps/userreports/reports/filters/values.py [hunk 3]
```diff
@@ -229,6 +229,12 @@ def _is_empty(self):
         operator = self.value.get('operator') or '='
         return self._has_empty_value() and operator == '='
 
+    def _is_exists(self):
+        """
+        Returns true if the value should be treated a filter to show non-empty data
```
Comment (millerdev) [inline corehq/apps/userreports/reports/filters/values.py:234]: nit: typo
```suggestion
       Returns true if the value should be treated as a filter to show non-empty data
```

File: corehq/apps/userreports/reports/filters/values.py [hunk 4]
```diff
@@ -229,6 +229,12 @@ def _is_empty(self):
         operator = self.value.get('operator') or '='
         return self._has_empty_value() and operator == '='
 
+    def _is_exists(self):
```
Comment (millerdev) [inline corehq/apps/userreports/reports/filters/values.py:232]: does "is exists" mean the same as "exists"?

**Review by millerdev — COMMENTED** [review]
Few nitpicky comments that are fine to ignore if you prefer.

File: corehq/apps/userreports/tests/test_report_filters.py (hunk 2 above)
Comment (czue) [reply corehq/apps/userreports/tests/test_report_filters.py:433]: I also wondered this, though it seemed unnecessarily complicated (also the OR should be an AND which I figured out in testing). Added comments in https://github.com/dimagi/commcare-hq/pull/28363 as a middle ground.

File: corehq/apps/userreports/reports/filters/values.py (hunk 1 above)
Comment (czue) [reply corehq/apps/userreports/reports/filters/values.py:227]: https://github.com/dimagi/commcare-hq/pull/28363

File: corehq/apps/userreports/reports/filters/values.py (hunk 4 above)
Comment (czue) [reply corehq/apps/userreports/reports/filters/values.py:232]: i think of it as meaning "does this look like an "exists" filter?" which is a little different. it also matches the `_is_xx` naming convention of the rest of the class

**Pull request merged**